### PR TITLE
Fixes image constructor with a filename as argument, solves #3947

### LINKED ISSUE #3947: Image.new(filePath) does not work on develop
Environment
-----------

SC 3.10 beta, develop branch, Mac OS 10.12.6 
commit 0f779a492519bef467c58fac9af3e37fe9e21e83

Expected Behavior
-----------------

```supercollider
a = SCDoc.helpSourceDir +/+ "images/Swamp.png";
i = Image.new(a);
```

should return an Image

Current Behavior
----------------

```supercollider
a = SCDoc.helpSourceDir +/+ "images/Swamp.png";
i = Image.new(a);
```

brings 

```
ERROR: Message 'extension' not understood.
RECEIVER:
Instance of String {    (0x124c5edd8, gc=4C, fmt=07, flg=00, set=03)
  indexed slots [126]
      0 : /
      1 : U
      2 : s
      3 : e
      4 : r
      5 : s
      6 : /
      7 : g
      8 : e
      9 : o
     10 : f
     11 : f
     12 : r
     13 : o
     14 : y
     15 : /
     16 : d
     17 : e
     18 : v
     19 : /
     20 : s
     21 : u
     22 : p
     23 : e
     24 : r
     25 : c
     26 : o
     27 : l
     28 : l
     29 : i
     30 : d
     31 : e
    ...
}
ARGS:
CALL STACK:
	DoesNotUnderstandError:reportError
		arg this = <instance of DoesNotUnderstandError>
	Nil:handleError
		arg this = nil
		arg error = <instance of DoesNotUnderstandError>
	Thread:handleError
		arg this = <instance of Thread>
		arg error = <instance of DoesNotUnderstandError>
	Object:throw
		arg this = <instance of DoesNotUnderstandError>
	Object:doesNotUnderstand
		arg this = "/Users/geoffroy/dev/supercol..."
		arg selector = 'extension'
		arg args = [*0]
	Meta_Image:open
		arg this = <instance of Meta_Image>
		arg path = "/Users/geoffroy/dev/supercol..."
	Meta_Image:new
		arg this = <instance of Meta_Image>
		arg multiple = "/Users/geoffroy/dev/supercol..."
		arg height = nil
		var ret = nil
	< closed FunctionDef >  (no arguments or variables)
	Interpreter:interpretPrintCmdLine
		arg this = <instance of Interpreter>
		var res = nil
		var func = <instance of Function>
		var code = "i = Image.new(a);"
		var doc = nil
		var ideClass = <instance of Meta_ScIDE>
	Process:interpretPrintCmdLine
		arg this = <instance of Main>
^^ The preceding error dump is for ERROR: Message 'extension' not understood.
RECEIVER: /Users/geoffroy/dev/supercollider/build/Install/SuperCollider/SuperCollider.app/Contents/Resources/HelpSource/images/Swamp.png
```

It seems that there was a change for opening images in this commit https://github.com/supercollider/supercollider/commit/ff3a0ccdf25666e022ad82cc9bfe93e6cf817f00

### PR DESCRIPTION
**Pull request by geoffroymontel**
Purpose and Motivation
----------------------

Fixes #3947

Types of changes
----------------

Bug-fix

Checklist
---------

- [X] All tests are passing
- [ ] If necessary, new tests were created to address changes in PR (and tests are passing)
- [ ] Updated documentation, if necessary
- [ ] This PR is ready for review

Remaing work
---------
I should have added a test for the Image constructor but I am not familiar enough with testing yet. I will try soon.

### CODE REVIEW

File: SCClassLibrary/Common/GUI/Base/QView.sc [hunk 1]
```diff
@@ -89,7 +89,7 @@ View : QObject {
 		this.background = color;
 	}
 
-	backgroundImage_ { arg image; this.setBackgroundImage(image) }
+	backgroundImage_ { arg image, tileMode=1, alpha=1.0, fromRect; this.setBackgroundImage(image, tileMode, alpha, fromRect) }
```
Comment (snappizz) [inline SCClassLibrary/Common/GUI/Base/QView.sc]: `setter_` methods should only take a single argument.

**Comment by geoffroymontel** [comment]
OK @snappizz ! I've reverted back my changes to the setter.
I will edit #3948 and #3952 accordingly

File: SCClassLibrary/Common/GUI/Base/QView.sc (hunk 1 above)
Comment (geoffroymontel) [reply SCClassLibrary/Common/GUI/Base/QView.sc]: fixed

File: .gitignore [hunk 2]
```diff
@@ -25,3 +25,6 @@ cmake-build-*
 
 *~
 *.orig
+
+# VS code
+.vscode
```
[the remark this answers is not in the document]
Comment (geoffroymontel) [reply .gitignore]: Would not it be useful for people ? I am browsing the code with Visual Studio Code and it is adding this .vscode file at the root of the project. And I don't want to push it to the repo.
As you wish !

[the remark this answers is not in the document]
Comment (brianlheim) [reply .gitignore]: @geoffroymontel  - @snappizz is right. Please avoid "drive-by" fixes. Also, please make sure your files end in a newline - it currently doesn't (you can see this in the diff view in the screenshot here). I believe git will complain if you try to commit a change with bad whitespace (trailing whitespace, missing final newline). If it doesn't on your machine, you should configure it so it does.

image

If you don't want to manually redo the commits it's possible to split commits using `git rebase` [`git help rebase`, "SPLITTING COMMITS" section]. I can help with that here, via email or slack.

File: SCClassLibrary/Common/GUI/Base/QView.sc [hunk 3]
```diff
@@ -89,7 +89,7 @@ View : QObject {
 		this.background = color;
 	}
 
-	backgroundImage_ { arg image; this.setBackgroundImage(image) }
+ 	backgroundImage_ { arg image; this.setBackgroundImage(image) }
```
Comment (brianlheim) [inline SCClassLibrary/Common/GUI/Base/QView.sc]: this change shouldn't be here.. it looks like all you really need to do for this PR is make the single change to the other line in `QImage.sc`. I would recommend making that change and force pushing to update this branch, then making the gitignore PR separately. If you want help with that I can give more advice here, via email or slack.

**Comment by geoffroymontel** [comment]
Thanks for your feedbacks @brianlheim and @snappizz, I used `git rebase -i` for the first time and cleaned up the history

**Pull request merged**